### ISSUE TITLE
[HL:S and HLD:S] - Game Breaking Bugs and Inconsistencies/Glitches

### ISSUE THREAD

**Issue by TheObscure** [ValveSoftware/Source-1-Games#1886]
Hi,

Many have already reported these problems elsewhere like on the steam forums, and they really need to be looked into as a lot of them are game breaking. I also reposted a couple of bugs in this report that was already brought up here in the bug list, but due to the fact that the last activity of an official Valve member fixing the reported things was more than a year and a half ago, I've felt that it is needed to bring everything up again along with new glitches in a nice and organized format that goes in great detail about each bug. So please don't hate on me for reposting things. I know that it is against the rules here, but I believe it is justified by the fact that as of yet we haven't got any official response that addresses this topic recently.

So let's get started:

---Singleplayer Only---

1 - The movement speed is ridiculously fast and the console command "sv_maxspeed" that was used to change/fix it back in the previous version, was removed. This renders the problem hopeless to fix for players themselves and having it like it is currently really breaks gameplay and immersion. To give a practical example of how this makes the game less enjoyable, in the singleplayer there are multiple puzzle areas that require the use of props and other map elements to cross a section, but now with the fast speed, you can easily bypass multiple areas with no regards to how the game should really be played. I would suggest that this command should be added back just like it was done in one of the betas for the MP version, so I can't think of a reason why this should not be added back to the singleplayer version as well.

2 -  Another set of commands were removed called "sv_accelerate" and "sv_airaccelerate" that were used to change movement related factors. This would be appreciated a lot in a fixed version of the singleplayer. These commands are currently present in the multiplayer version. I believe it got removed from the singleplayer version during the porting phase between two source engine builds.

http://i.imgur.com/8BIHyWB.jpg

3 -  On map c2a5c on the singleplayer, the skybox is missing and this is causing graphical issues.

http://i.imgur.com/myocMAF.jpg

http://i.imgur.com/LGXeGLR.jpg

4 -  Another set of artifacts that I encountered whenever I used the flashlight on the map called c2a4c. Here is a video showing the bug: 

https://www.youtube.com/watch?v=5vd9Ronv0ac

5 - In the control room right before you enter the test chamber, one of the scientists sitting in front of a computer is floating in the air above his chair.

http://i.imgur.com/5uQkq9p.jpg

6 - When a friendly NPC like a scientist or a security guard is hurt and survives any form of attack, he will continuously make hurt noises as if someone is still inflicting damage.

--Multiplayer Only---

7 - Every entity on multiplayer like ammo, weapons and even map entities such as elevators, doors, stationary weapons are doubled. To put it in better detail, there are basically two in one of everything. Instead of picking up one health pack, you will get two. This is a major problem, because this makes multiplayer matches strange and very ridiculous especially on the map Subtransit where there are now two trams instead of one. This is honestly a game breaking glitch. Here is an example of the problem.

http://i.imgur.com/gfaUEQ7.jpg

http://i.imgur.com/M7jwmQN.jpg

8 - The muzzleflashes of the MP5 are broken. After firing one round, the weapon will continuously flash even though you already have stopped firing. The muzzleflashes also appear when firing the attached grenade launcher which is not supposed to happen.

9 - The crowbar impact sounds are still halfway broken in the current beta of the multiplayer version. It will make the correct impact sounds on most surface materials, but on some like wooden surfaces and stationary health and armor charges the wrong sounds are playing instead. If this were to be looked into, please check out the way it is in the singleplayer version where it is fixed.

10 - The glock pistol still has 18 bullets in both the beta and standard version of the multiplayer version even though it says that it was fixed in one of the last betas.

http://i.imgur.com/xvuWC12.jpg

11 - On the multiplayer version it is no longer possible to change your player model. Every time you apply a new character model, it switches back to the "Gordon" player model as a default. This totally breaks team deathmatch for obvious reasons.

12 - The disconnect button is missing from the menu when in a loaded match on multiplayer. This is not the case on different Valve multiplayer games.

http://i.imgur.com/UOYWAGt.jpg

13 - The "barnacle" alien entity was removed from multiplayer. This caused problems on custom maps made by the community.

14 - It is no longer possible to load the maps of the singleplayer on the multiplayer. People did this so that they could play Half-Life Source cooperatively. Another thing is that the singleplayer NPCs in multiplayer are broken and don't attack the player or behave in the way they should. This breaks cooperative play even more.

15 - The multiplayer version is impacted by severe server lag issues when playing online. Especially when crouch jumping where the whole screen shakes as a result. People also complain about crash issues which render the game even more unplayable.

16 - At the weapon selection area where you can see which guns you have, the ammo indicator bars  should also be green like it is in the singleplayer version.

http://i.imgur.com/M0f6LcC.jpg

17 - When using the crossbow's zoom, the mouse's sensitivity doesn't change to a lower setting that makes it easier to use the zoom feature.

---Both Versions---

18 - All chrome textures are broken on weapons and on other models. This gives models a very unnatural look. This was fixed in the previous version of the game before the steampipe engine changes. 

http://i.imgur.com/Qs02INA.jpg

19 -  All the areas in the game where you have to use the bouncers to get to higher places, instead of giving you a little boost in the air, they launch you at a very high velocity making it impossible to land on the ground without losing your life. This is a very big problem in the Xen chapters where you will have to rely on these things to continue with the campaign. This is even more of a problem on multiplayer servers on the map Bounce where there are a lot of these things.

The first screenshot shows the maximum height at which you get launched on a Xen map in the singleplayer (This is serious, and I didn't use noclip):

http://i.imgur.com/qUZbE4F.jpg

The following screenshots show the maximum height at which you get launched at on the Bounce map. The first screenshot is Half-Life Source, the second one is on normal Half-Life.

http://i.imgur.com/rF0xUDw.jpg

http://i.imgur.com/WoK8jT4.jpg

If I have to guess, then the pushing force should be decreased by about 50% to make it accurate towards normal Half-Life.

20 - All the weapons in the game have very slow and unnaturally recoil, where in the original Half-Life it was very fast and responsive and allowed for better aiming while shooting at longer distances. I won't consider this a bug, but it is a very big inconsistency and it really makes the gameplay much different in a terrible way compared to how it is in normal Half-Life.

21 - The shotgun is supposed to have 2 firing sounds that accommodate both firing types. A separate sound for the single fire, and also a separate double fire sound. In Half-Life Source, the former is used for both attacks. The shotgun's pump sound is also out of sync with the double fire's animation, which was not present in the original Half-Life. On previous versions of Half-Life Source, the game was using both the sounds, but that was before the steampipe updates.

22 - When having the Glock pistol equipped, the knuckle part of the glove has a red color for the HEV suit details, where it is supposed to be orange like it is with all the other weapons.

http://i.imgur.com/ULqDO7G.jpg

http://i.imgur.com/xvuWC12.jpg

23 - The snark weapon's world model is wrong. It should be a nest model, not the model of an individual snark.

24 - The shell ejection of the Glock pistol has an offset which makes it look unrealistic. This was not a bug on normal Half-Life.

http://i.imgur.com/X7AQpwn.jpg

---Feature Requests---

1 - We would like to have the title menu of Half-Life Deathmatch: Source in such a way that it says "Deathmatch" somewhere. This will make the game feel more complete and official, and not like a mod where content feels incomplete and inconsistent.

2 - We would also appreciate it if Valve release the SDK for both Half-Life Source and Half-Life Deathmatch: Source. I feel that this could be done, because the SDK for the original Half-Life was released, and as a result many mods had popped up and still do to this day. Having the SDK of these two games would be crucial to have when a modder wants to port a standard Half-Life mod over to Half-Life Source without having the need to recode almost the entirety of the game. This will as a result give another good incentive to players to buy and play Half-Life Source again, and it is a well-known fact that there is a correlation between the popularity of a game and the availability of modding support.

---Final Thoughts---

As a long time fan of the Half-Life series, I really hope that these issues can get resolved so that the games can have a much better and consistent gameplay like it was in the original Half-Life and on the previous versions of Half-Life Source and Deathmatch Source. Overall both of these games were much better off before the steampipe update in terms of how many bugs were present, but with a little more attention from the developers, these two games can be restored back to its former glory, or even better if the developers could fix even those bugs that were present before the steampipe update. I also noted a lot of discontent on the Steam forums about all these bugs. Many people are complaining and request fixes for this so that both games can be more playable and stable, especially the multiplayer version that is practically unplayable now. The community also feel that both the SP and MP versions should not be left in the cold, and should receive more treatment like Half-Life 2 which is still getting fixes even though it is already 11 years old. We are not speaking of continuous updates, but instead for only those bugs that are put forward here above in a detailed manner. There might be more bugs that the community and I have overlooked, but we will post them here as soon as we hear of them, and we would also be glad to take part in a future beta if it were possible to have one. Again, we really would like to have these two games fixed, because we payed $10.00 for each of them and as Steam customers we deserve to get the product that we payed for and for how it was advertised.

Lastly, I did extensive research into all of these issues listed above in my report so that I can confirm them properly, and it took me quite a while to formulate this report so in return I expect to hear from you developers.

Thank you and best regards,

TheObscure

**Comment by GrO** [comment]
Very good job TheObscure!!! At last someone did it, you're the best!
Maybe Valve will fix it, but atm they're loving they're csgo-money-maker (I didn't buy CSGO btw).

**Comment by TheObscure** [comment]
Thanks GrO! 

I just hope that all my effort in compiling this list was not in vain. If I look at the status of all the other opened issues for Valve games and how they are still not resolved, I fear that this may end up being never even read by one of their developers. The last time an issue was reviewed here on this site was more than a year ago. :(

But yeah, at least we should try something to get these two games properly fixed. Maybe I should email this to Valve so that I can get a direct response from them. :/

**Comment by 1157** [comment]
@TheObscure  Good report. SDK is included with hldms  just needs to be configured and a few files
moved around. I will post if your interested.

**Comment by TheObscure** [comment]
@1157 Thank you. I just thought by myself that someone had to say something about these two games that have been mistreated and then left to rot. The worst part is that Valve is still selling them for $10 each, which is really sad considering that both are broken and unplayable. Let's hope that they even read this.

As for the SDK, are you saying that the latest steampipe version has the SDK for it included? I only know about the old leaked SDKs for it, but that version is pre-steampipe. :/

**Comment by 1157** [comment]
Prior to "Steampipe" hldms and hls ran on the 2006 ep1 engine. With Steampipe we also got a new engine the 2013 MP engine.So now we are with everybody else (CSS,HL2MP, ect).
Here is the info for the SDK. Map's compile much,much faster.

http://www.hldms1157.com/Hammer/hl1mp_2013_hammer.zip

**Comment by Rastrelly** [comment]
Some of those bugs I didn't experience, but overall I think this report is valid.

**Comment by Redeemer6666** [comment]
Yeah this really needs sorting out, I've encountered nearly if not all the bugs described above. Please fix this game Valve it really does deserve some love.

**Comment by Redeemer6666** [comment]
@1157 How'd you configure HLS for the SDK I could really use a hand here, thanks if you can this has really been kicking my ass lol

**Comment by TheObscure** [comment]
@Redeemer6666 

Yes, I know that feeling of buying both games and then expecting them to be at least as good as the normal Half-Life. 

For anyone reading this that just bought HLS and HLDS, you are in for a big surprise.

If you read this and haven't bought these games, then my recommendation would be to not buy it. Rather go with the default Half-Life.

**Comment by Redeemer6666** [comment]
@TheObscure I'm with you, these games are utter crap in their current state and not worth any money whatsoever. Good job I brought the Anthology but even standard Half-Life on Steam has stupid OpenGL problems so I had to pirate a previous version off the web that contained D3D support and create a "hybrid" version just to be able play it. Valve are being really stupid with their lack of support because I had to STEAL a game just to get it working!!! Tbh I knew I was in trouble with the Steam version 5 minutes in, When I saw the scientists were floating above their chairs. Like I said on a previous post on the steam page funny thing is I'd fix it, if they'd have just released a proper SDK.

**Comment by 1157** [comment]
@Redeemer6666   here you go

http://www.hldms1157.com/Hammer/hl1mp_2013_hammer.zip

There are "workarounds" for most broken things like "func_door" (using func_movelinear),

**Comment by Redeemer6666** [comment]
@1157 Truth is I'm looking at adding things for a mod, Aim was to create some mad SMOD type thing adding Opposing Force weapons and such.
Guessing that can't be done without getting at the games .CPP files?

**Comment by TheObscure** [comment]
@Redeemer6666 Do you know about the leaked SDKs for both HLS and HLDMS? They are on the same current steampipe engine version if I'm right. You can use them if you are interested in fixing some of the bugs or doing that smod thing. There is even this guy on the moddb site that is porting both Blue Shift and Opposing Force to the latest source engine. You might get the SDKs from him.

http://www.moddb.com/mods/half-life-opposing-force-source

**Comment by Redeemer6666** [comment]
@TheObscure There's a leaked SDK? That's news to me, Thanks buddy I've checked out his page sent him a message and we'll see what he says. Hope I can get access to that SDK I'll fix what I can and post it onto Steam forums. Somethings gotta be done and us fans are the only ones interested apparently.

**Comment by TheObscure** [comment]
@Redeemer6666 Yes there is one (for both MP and SP I think). And no problem mate. I just want to know at the end of the day that I've done everything in my power to make sure we get the HLS and HLDMS that we all deserve. Just imagine if people in Valve had the same mindset than me regarding these two games. Those would've been fixed a long time ago if that were the case.

Regarding all those bugs I've mentioned in my report, a few of them can be fixed with other means. For example the two glitches where you can't see the skybox and the other one with the flashlight artifacts, those can be fixed by editing the maps involved I believe. As for the rest, I think it's up to a coder to fix them. :/

So you say that you have the experience to tackle this issue and to fix some of the bugs? If yes, then I think it is time that we get these games sorted out. Unfortunately I am not a Source engine modder, so I can't be of any assistance. I am only modding Battlefield 2 and I am a 2D graphic artist and a basic modeller, so if those specific skills are needed, I'd be glad to help out. But coding is not my forte. 

Also, while I was thinking about 2D graphics, an idea came to me. Do you think it would be possible to make bumpmap textures for the normal Half-Life Source map textures? I think adding this to the game can give better graphics to the game while not changing anything drastically. And maybe redoing the whole HD pack that was included with the steampipe would be something that I'd be interested in. I do know a few things about the Source and GoldSRC engines as I've tinkered a bit with a few things, so for me it won't be a case of learning everything from the beginning. 

I guess we can assemble a small team to do all these things?

**Comment by Redeemer6666** [comment]
@TheObscure I haven't managed to get hold of the creator of that Opposing Force Source mod just yet but I have manged to find (from some shady Russian website) what I believe are the .cpp files contained within the server/client dlls.
Now these are from a previous leaked 2007 Source release so I don't know how useful these files will be or if anythings missing because as I haven't had a good look at em yet. I'll leave a link at the end of my post for anybody interested though.

While I'm not exactly new to Source modding, I wouldn't quite claim I'm "experienced". What I do have however is a shit tone of free time on my hands and the will to learn whatever if necessary to get the job done lol 
I've been wanting a good opportunity to expand upon what I know about the Source engine and this seems like a perfect time and game to attempt this on.
So I'm all in on creating a loose team of people trying to fix this poor and abandoned gem, it seems like a bloody good idea.

Any yeah I think some new bumpmaps could be made and supported, I'm sure I've already seen somebody do this so it shouldn't be too difficult.
I'm already in the process of sorting out Gearbox's original "HD" pack and porting it to Source, like you said the current one is absolutely woeful (why the hell is the pistol SILVER lol)  

Anywho if anyone fancies poking around in these files too, Here they are.

https://mega.co.nz/#!xFIgEDqI!b9jHXwruhBnYTp9FZGN03X3Pl9tnDK3nr-Qw12Xs-wk

**Comment by TheObscure** [comment]
@Redeemer6666  If you can't get the SDK stuff from the creator of those two ports, then instead ask this guy on steam:

http://steamcommunity.com/id/itsa2by4

He told me that he has the full SDKs of not only the latest version, but also of previous ones. 

Also, it would be fantastic if you can lend your coding expertise to help on fixing this game, and since you have a lot of free time on your hands I guess this makes it more awesome!

As for adding bumpmaps, I really think this would be a great addition since it is according to you 100% possible. This will update the game's graphics without changing the overall feeling of the game, and I am very much against mods/addons that ruin the nostalgic look of the good ol' Half-Life. Also, since I am going to have to learn a bit to work with the Source engine's vmt and vtf files again, I guess I can try to fix those shiny and glossy metal textures too. I think they look horrible, especially in the steampipe version. I assume that can be fixed too right? 

About the porting of the HD pack. I think you should pause with that and rethink exactly what you want to port. Yes, I know the HD pack from Gearbox is better than the default pack, but there are a lot of newer and better upgrades for them. I don't know whether you have seen the PlayStation 2 version's models yet, but if you haven't then you should rather take a look at them. All of them have been ported and re-updated to look even better than the original HD models. Take especially a good look at the knuckle part of the v_models. They are much higher poly as opposed to a flat texture:

http://hl.gamebanana.com/skins/29335

There are many upgrades for the classic HD pack, and by looking around on that site, we might find some better models for the HD pack's other models too.

Lastly, do you have anything to do with this steam group that popped up today? I can't find you on the member list, and it is funny how the group started just moments after I suggested a group of modders to fix HLS and HLDMS lol. 

http://steamcommunity.com/groups/hl1s_update#

**Comment by Theuaredead** [comment]
@TheObscure I wasn't even aware it was mentioned here, I was working on organizing a mod/update group for the past week.

**Comment by Theuaredead** [comment]
@Redeemer6666 I recommend snipping that Source Code leak, Valve will actually notice that and take down both the link and probably try to get you banned, Valve dislikes people touching leaked code.

**Comment by TheObscure** [comment]
@Theuaredead If I may ask, are we going to base this project on the Steampipe version using those leaked SDKs?

**Comment by Theuaredead** [comment]
That leaked SDK is running on Source Base 2007 (which, is actually where some of the Steampipe bugs came from, because either they never converted it to work with the Engine, or they used an older code base), But I do plan on upgrading it to Steampipe aslong as the major issues can be ironed out, plus I want to make it so we have a lot of the bugs cleared up on the Source Base 2007 version so we don't end up with a nearly 1:1 Steampipe codebase

**Comment by TheObscure** [comment]
Ah I see, so you think all of this is doable; I mean porting it to the Steampipe version whilst having it at the end of the day like it should have been on the steam version? Another concern I have; how many coders do we have so far for this project, because as I understand, this project will mainly rely on them right?

**Comment by Theuaredead** [comment]
Yeah, the project will rely on mappers, coders, and those who can convert models from GoldSource to Source (me! fun fact: alfred asked me to email him about helping out with this update project, but he never replied back)

We currently have 1 programmer (maybe 2?), but if we get the attention of people, more and more people could start helping out.

I want the project to be how Half-Life: Source should of shipped (most people assume it was supposed to be like CSS, where its an HD remake of HL1, when it's not)

**Comment by Theuaredead** [comment]
@TheObscure I replied to your comment on the Steam page, fyi

**Comment by TheObscure** [comment]
If we can get as much coders as possible, then I guess we are off to a good start. Mappers will be our second priority I think, since that isn't the focus of the HLS and HLDMS part of the project. They will be, however, much needed when we want to start on DMC and TFC.

("I want the project to be how Half-Life: Source should of shipped (most people assume it was supposed to be like CSS, where its an HD remake of HL1, when it's not)")

Yes, I feel the same way. People can play Black Mesa if they are interested in modern graphics, but for me I prefer HLS all the way. I really think we can make this project work. We already have the enthusiasm, so all we need now is a bunch of coders to help us! :) 

"(alfred asked me to email him about helping out with this update project, but he never replied back)"

Huh, so you are saying that he is interested in this then? Well if you didn't send that email too long ago to him, you might expect to hear from him on a later stage I guess...?

**Comment by Theuaredead** [comment]
@TheObscure DMC's maps are actually already on the Source Engine, along with an earlier version of CSS (where it was once like HL1S)
and that Alfred thing, that was from 2013, on the GitHub page. https://github.com/ValveSoftware/Source-1-Games/issues/1473

**Comment by TheObscure** [comment]
Oh I see, so I guess we can't count on his help. Also, I am glad that we got a head start with the already ported maps of DMC. This leaves the project with less work which is good.

Lastly, will this project be a decline of Valve's copyrighted stuff, or will this be okay as long as we don't involve money for this? I mean, the original TFC and DMC are still commercial games on steam, so that is where I get my concern from.

**Comment by Theuaredead** [comment]
Valve doesn't mind when people port GoldSource to Source or Source to GoldSource, they care when people use stolen code, and to my knowledge, Valve hasn't shut down any mod that has used stolen code. Valve is still friendly to modders, even if it's not as friendly as they used to be.

**Comment by Redeemer6666** [comment]
@TheObscure Yeah the HD pack in question was just a little side project of mine, I just thought it might be cool to recreate Gearboxes official PC Hi def pack for purists and such. I was going to use the updated PS2 models (with some tinkering) in my own mod, I quite fancied porting over those fancy animated HEV chargers and Med stations from Decay too. But that's enough about my own crap for now lmao 

Guessing you've probs already figured out I'm not part of that new Steam group (scrolls discussions) but I'd sure like to be, like I said in my previous post I'm no expert but I'm willing and want to learn new things.

**Comment by Theuaredead** [comment]
@Redeemer6666 I actually ported the PS2, DC, and PC models for use in HL1S for Valve

They didn't bother patching any of them in.

https://github.com/ValveSoftware/Source-1-Games/issues/1471

**Comment by Redeemer6666** [comment]
@Theuaredead Thanks for the heads up, I've removed the link, I was being a tad silly there huh? 
Anyways is cool to go ahead and join your Steam group, I'm not massively skilled but I am a spare pair of hands and I'm learning all the time :) 

About these SDKs, what exactly are we using and if possible can I grab a link? Its probs best we all work using the same files lol 

And oh nice, is there anywhere I could grab them models from too?

**Comment by Theuaredead** [comment]
@Redeemer6666 the files got deleted off of puush a while back, I got the files backed up on my computer still, they're a bit buggy, I could probably do better.

We're using the Source Base 2007 leak, which is what you posted (i don't remember if you posted the full version or not)

**Comment by Theuaredead** [comment]
Realization kicks in that this is the most popular thread in this GitHub, and its now about how we plan to go against this update.

**Comment by Redeemer6666** [comment]
@Theuaredead Not sure if I have the full SDK like I said I just stumbled upon it on some Russian website, translations were not the best lol so I'm really not too sure if I got all HLS stuff. The SDK is about 1gb in size if that helps you identify it :D 

Haha yep sometimes you just have to take a step back to figure out what the hell you did wrong, Shame Valve never tried this themselves XD

**Comment by Theuaredead** [comment]
@Redeemer6666 They officially stopped caring about HL1S in 2005, the only reason it was ported over to Source Base 2013 and Steampipe was because they were removing all of the old downloading system.

Also, I checked the mega link you posted, its the full leak.

Also, preferably we should all probably run off to the HL1S Update GitHub, since I don't think we should confuse lurkers thinking we're working with Valve on this. Don't know how we'll pick up from the discussion, but bug posts should be posted on it. 

https://github.com/Theuaredead/Half-Life-Source-Update

**Comment by Redeemer6666** [comment]
That's cool, righto i'm off to work then. Thanks for all the help guys :) If we need to talk perhaps the steam page will do.

**Comment by kisak-valve** [comment]
Hello @TheObscure,

This is the wrong way to use issue trackers. There should only be one issue per issue report. As such, only the first 2 issues requesting sv_maxspeed, sv_accelerate, and sv_airaccelerate be re-added will be tracked in this issue report.

Partial list of issues this is issue is a duplicate of:
Issue 3 is being tracked at #1326.
Issue 5 is being tracked at #1346.
Issue 6 is being tracked at #1218.
Issue 7 is being tracked at #1368.
Issue 8 is being tracked at #1419.
Issue 10 appears to be fixed.
Issue 11 is being tracked at #1427.
Issue 12 appears to be fixed.
Issue 18 is being tracked at #1724.
Issue 21 may be related to #1489.
Issue 22 is being tracked at #1251.
Issue 23 is being tracked at #1398.

In the future, after searching the existing issue reports, please open separate issue reports, one issue per issue report so that the individual issues can be tracked properly.
